acceptance/tests/environment/use_enc_environment.rb

```diff
@@ -11,21 +23,43 @@
 END
 on master, "chmod 755 #{testdir}/enc.rb"
 
+end
+
+apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+  File {
+    ensure => directory,
+    mode => "0770",
+    owner => #{master.puppet['user']},
+    group => #{master.puppet['group']},
+  }
+  file {
+    '#{testdir}/environments':;
+    '#{testdir}/environments/production':;
+    '#{testdir}/environments/production/manifests':;
+    '#{testdir}/environments/special/':;
+    '#{testdir}/environments/special/manifests':;
+  }
+  file { '#{testdir}/environments/production/manifests/site.pp':
+    ensure => file,
+    mode => "0640",
+    content => 'notify { "production environment": }',
+  }
+  file { '#{testdir}/environments/special/manifests/different.pp':
+    ensure => file,
+    mode => "0640",
+    content => 'notify { "expected_string": }',
+  }
+MANIFEST
+
 master_opts = {
-  'master' => {
-    'node_terminus' => 'exec',
-    'external_nodes' => "#{testdir}/enc.rb",
-    'manifest' => "#{testdir}/site.pp"
+  'main' => {
+    'environmentpath' => "#{testdir}/environments",
   },
-  'special' => {
-    'manifest' => "#{testdir}/different.pp"
-  }
 }
-
-create_remote_file(master, "#{testdir}/different.pp", 'notify { "expected_string": }')
-
-on master, "chown -R #{master['user']}:#{master['group']} #{testdir}"
-on master, "chmod -R g+rwX #{testdir}"
+master_opts['master'] = {
+  'node_terminus' => 'exec',
+  'external_nodes' => "#{testdir}/enc.rb",
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.